### PR TITLE
Connection cache improvements

### PR DESCRIPTION
**Pull request by vpodzime**
This is a follow-up for #3515. See individual commits for details.

### CODE REVIEW

**Comment by vpodzime** [comment]
@cf-bottom jenkins with exotics, please

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/2199/)

(with exotics)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/2199/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-2199/

File: libcfnet/conn_cache.c [hunk 1]
```diff
@@ -105,26 +116,27 @@ AgentConnection *ConnCache_FindIdleMarkBusy(const char *server,
...
-            }
-            else if (svp->status == CONNCACHE_STATUS_OFFLINE)
-            {
-                Log(LOG_LEVEL_DEBUG, "FindIdle:"
-                    " connection to '%s' is marked as offline.",
-                    server);
-            }
-            else if (svp->conn->conn_info->sd >= 0)
+            Log(LOG_LEVEL_DEBUG, "FindIdle:"
+                " connection to '%s' seems to be busy.",
```
Comment (amousset) [inline libcfnet/conn_cache.c]: It may not be a connection to `server` as connection has not been matched yet.

Comment (amousset) [inline libcfnet/conn_cache.c]: In `BUSY`, `OFFLINE` and `BROKEN` cases, the connection may not be a connection to `server` as it has not been compared yet with these changes.

Comment (vpodzime) [reply libcfnet/conn_cache.c]: Good catch, thanks!

**Comment by vpodzime** [comment]
@cf-bottom another jenkins with exotics, please

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/2208/)

(with exotics)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/2208/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-2208/

**Comment by olehermanse** [comment]
Thanks for helping review this, @amousset! 🎉

**Pull request merged**